wikidict/utils.py

```diff
@@ -485,22 +484,33 @@ def _convert_math(expr: str, packages: List[str] = []) -> str:
     expr = expr.replace("\\infin", "\\infty")
     expr = expr.replace("\\rarr", "\\rightarrow")
 
+    dvioptions = ["--no-fonts"]
+    with BytesIO() as buf_svg:
+        preview(
+            f"${expr}$",
+            output="svg",
+            viewer="BytesIO",
+            outputbuffer=buf_svg,
+            dvioptions=dvioptions,
+            packages=tuple(packages),
+        )
+        buf_svg.seek(0)
+        raw_svg = buf_svg.read()
+
     dvioptions = ["-T", "tight", "-z", "0", "-D 150", "-bg", "Transparent"]
-    with BytesIO() as buf, BytesIO() as im:
+    with BytesIO() as buf_png:
         preview(
             f"${expr}$",
             output="png",
             viewer="BytesIO",
-            outputbuffer=buf,
+            outputbuffer=buf_png,
             dvioptions=dvioptions,
             packages=tuple(packages),
         )
-        Image.open(buf).convert("L").save(im, format="gif", optimize=True)
-
-        im.seek(0)
-        raw = im.read()
+        buf_png.seek(0)
+        raw_png = buf_png.read()
 
-    return f'<img style="{IMG_CSS}" src="data:image/gif;base64,{b64encode(raw).decode()}"/>'
+    return f'<img style="{IMG_CSS}" src="data:image/svg+xml;charset=utf8,{parse.quote(raw_svg)}" data-png="data:image/png;base64,{b64encode(raw_png).decode()}"/>'  # noqa
```

Comment:
Do we have SVG CSS to include too? cc @Moonbase59

Reply:
I am mostly talking about dark theme, is it working out of the box with SVG?

Reply:
> Do we have SVG CSS to include too? cc @Moonbase59

Not necessarily, the latest SVG I create with the new Python module have a `style` attribute and thus "handle themselves". The only CSS needed with that construct would be a color `fill` (in CSS) in case a reader (non-e-ink) wants night mode. Possible e-readers using e-ink will inverse b/w via their drivers.

Reply:
Why would we need PNG in .df and StarDict? StarDict works fine here with SVG. Let the readers adopt… ;-) Guess @Frenzie and KOReader will be the first.

Reply:
Sidenote, but what I'm seeing here is neither SVG nor PNG but data URI. That's an additional complication that wastes a lot of space to boot.

> Guess @Frenzie and KOReader will be the first.

Maybe, maybe not. I don't have time to look into upgrading MuPDF just for this, unless it doesn't have a weird viewBox. I did however post a dumb sample workaround here: https://github.com/koreader/koreader/discussions/8777 and if I find the time I might potentially patch MuPDF itself more properly.

Reply:
@BoboTiG: To clarify on night modes, the CSS would usually be done in the reader’s theming, like in a GoldenDict nightmode theme, you could simply state:
```css
svg {
  fill: white;
}
```

@Frenzie: I still hope we’ll eventually decide in favour of "pure SVG" (no `img`, no base64), because that’s what my `latex2svg` generates and seems to work pretty well, see https://github.com/BoboTiG/ebook-reader-dict/issues/1198#issuecomment-1030839050. This kind is also the only one that can be very easily CSS-styled.

Reply:
> Why would we need PNG in .df and StarDict? StarDict works fine here with SVG. Let the readers adopt… ;-)

Seriously ? My first though is always with my fellow Kobo users, so I care a bit less about StarDict but still. 
It seems extremely harsh  give users a dictionary and then tell them, "Sorry, you don't see the images, go find your developers and shout" (Now that I wrote it, it's not extremely nice to these developers either)


> I still hope we’ll eventually decide in favour of "pure SVG" (no `img`, no base64)

I will test again but just outputing the SVG as it didn't work last time I checked. I don't really care about styling it it works on Kobo and if PNG is used for other users (these users will definitely don't care about CSS styling... if they can't see pictures at all in the first place...)

Reply:
Well, I gave you a Kobo test dictionary and screen shots in https://github.com/BoboTiG/ebook-reader-dict/issues/1198#issuecomment-1030839050 and it seems to render fine on Kobos using "pure" SVG.

Reply:
Am I right when I say inline SVGs are working on Kobo, and StarDict, and so we could simple generate SVG for everyone?

Then, maybe could we use the latex2svg (https://github.com/BoboTiG/ebook-reader-dict/issues/1198#issuecomment-1039631349)?

Reply:
My understanding is that with latex2svg we have a working solution for Kobo and inline SVG indeed. But for StarDict, it's far from clear. @Moonbase59 suggested to go with it and let the (e-)readers adapt. I don't think it's a good idea but hell, I'm not a stardict user... except for Koreader on Kobo for testing purpose.

If we wanted to add another output format, like Tolino... it would bite us again... Apparently Tolino still use a very old dictionary format. They didn't updated it to a new format. I don't really see why they would care to add SVG support.

Reply:
I do not want Tolino to be a blocker.
@Moonbase59 did you find time to test SVG on it?

Reply:
Kindly ping @Moonbase59 :)